### ISSUE TITLE
Question signed sqrt normalization

### ISSUE THREAD

**Issue by zhujiem** [HaoMood/blinear-cnn-faster-#2]
Is it correct to use torch.sqrt(X + 1e-5) since X may have negative values? Is there any reason for this simplication? Thanks!

https://github.com/HaoMood/blinear-cnn-faster/blob/master/src/model.py#L117

```
# Normalization.
# X = torch.sign(X) * torch.sqrt(torch.abs(X) + 1e-5)
X = torch.sqrt(X + 1e-5)
X = torch.nn.functional.normalize(X)
```

**Comment by HaoMood** [comment]
Hi, thanks for your comment.

Since X is obtained from the relu5-3 activation of VGG, it is nonnegative elementwise. The bilinear result is nonnegative as well. Therefore, multiplying the sign or not does not affect the result.

Best regards,

Hao
